src/views/News/index.js

```diff
@@ -18,113 +23,177 @@ import {
   DialogActions,
   DialogContent,
   DialogTitle,
+  DialogContentText,
   MenuItem,
 } from '@material-ui/core';
 import CloseIcon from '@material-ui/icons/Close';
 import { ReactComponent as NoConnectionImage } from 'NoConnection.svg';
-// testing for future feature to upload image
-// import IDUploader from '../IDUploader';
-// import Dropzone from 'react-dropzone';
-
-const styles = {
-  searchBar: {
-    margin: '0 auto',
-  },
-  newNewsForm: {
-    backgroundColor: '#fff',
-  },
-  fab: {
-    margin: 0,
-    top: 'auto',
-    right: 40,
-    bottom: 40,
-    left: 'auto',
-    position: 'fixed',
-    zIndex: 1,
-  },
-};
 
-export default class StudentNews extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      open: false,
-      search: '',
-      openPostActivity: false,
-      loading: true,
-      categories: [],
-      news: [],
-      personalUnapprovedNews: [],
-      filteredNews: [],
-      network: 'online',
-      newPostCategory: '',
-      newPostSubject: '',
-      newPostBody: '',
-      snackbarOpen: false,
-      snackbarMessage: 'Something went wrong',
-      currentUsername: '',
-      // false if not editing, newsID if editing
-      currentlyEditing: false,
+const BREAKPOINT_WIDTH = 540;
+const CROP_DIM = 200; // pixels
+
+const StudentNews = (props) => {
+  const [search, setSearch] = useState('');
+  const [openPostActivity, setOpenPostActivity] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [categories, setCategories] = useState([]);
+  const [news, setNews] = useState([]);
+  const [personalUnapprovedNews, setPersonalUnapprovedNews] = useState([]);
+  const [filteredNews, setFilteredNews] = useState([]);
+  const [network, setNetwork] = useState('online');
+  const [newPostCategory, setNewPostCategory] = useState('');
+  const [newPostSubject, setNewPostSubject] = useState('');
+  const [newPostBody, setNewPostBody] = useState('');
+  const [showCropper, setShowCropper] = useState(null); //null if no picture chosen, else contains picture
```

Comment:
I've been thinking during this whole project that `showCropper` is pretty bad. I never changed it because hopefully in the future, some of these image-related functions and JSX bodies will be moved to a separate file; additionally, `showCropper` _does_ determine whether or not the cropper is shown. But what do you think of `cropperImageData`? I use phrases like "image data" often around the project because I feel like it's more accurate.

Reply:
cropperImageData seems like a much better name to me.

Reply:
This looks like it still needs to be done.

Reply:
Done.